src/DraupMembershipERC721.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import {IERC2981, IERC165} from "openzeppelin-contracts/contracts/interfaces/IERC2981.sol";
+import {DefaultOperatorFilterer} from "operator-filter-registry/src/DefaultOperatorFilterer.sol";
+import {ERC721} from "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
+import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {MerkleProof} from "openzeppelin-contracts/contracts/utils/cryptography/MerkleProof.sol";
+import {PaddedString} from "draup-utils/src/PaddedString.sol";
+import {IRenderer} from "./IRenderer.sol";
+
+contract DraupMembershipERC721 is ERC721, Ownable, DefaultOperatorFilterer {
+    uint256 public immutable MAX_SUPPLY;
+    uint256 public immutable ROYALTY = 750;
+    bool public transfersAllowed = false;
+    IRenderer public renderer;
+    string public baseTokenURI;
+    uint256 public nextTokenId;
+    bytes32 public merkleRoot;
+
+    // Mapping to track who used their allowlist spot
+    mapping(address => bool) private _claimed;
+
+    constructor(uint256 maxSupply, string memory baseURI) ERC721("Draup Membership", "DRAUP") {
+        MAX_SUPPLY = maxSupply;
+        baseTokenURI = baseURI;
+    }
+
+    error InvalidProof();
+    error AlreadyClaimed();
+    error MaxSupplyReached();
+    error TransfersNotAllowed();
+
+    function mint(bytes32[] calldata merkleProof) public {
+        bytes32 leaf = keccak256(bytes.concat(keccak256(abi.encode(msg.sender, 1))));
+        if (!MerkleProof.verify(merkleProof, merkleRoot, leaf)) {
+            revert InvalidProof();
+        }
+        if (_claimed[msg.sender]) {
+            revert AlreadyClaimed();
+        }
+        _claimed[msg.sender] = true;
+        nextTokenId++;
+        if (nextTokenId > MAX_SUPPLY) {
+            revert MaxSupplyReached();
+        }
+        _mint(msg.sender, nextTokenId - 1);
+    }
+
+    // token trading is disabled initially but will be enabled by the owner
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 firstTokenId,
+        uint256 batchSize
+    ) internal virtual override {
+        if (!transfersAllowed && from != address(0)) {
+            revert TransfersNotAllowed();
+        }
+        super._beforeTokenTransfer(from, to, firstTokenId, batchSize);
```

Comment:
It doesn't seem necessary to call this base method. The base method's logic is only executed if `batchSize > 1` which is never the case here.